File: Python/isHappy.py
def isHappy(n):
    """
    :type n: int
    :rtype: bool
    """    
    di = []
    while n not in di:
       di.append(n)
       s = 0
       while n > 0:
           r = n % 10
           s += r*r
           n = n / 10
       if s== 1:
           return True
       else:
           n = s    
    return False
     
print isHappy(7) 

File: JavaScript/isHappy.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
    var arr = [],
        sum = 0;
    if (n === 0) return true;

    while (sum !== 1) {
        while (n !== 0) {
            sum += Math.pow(n % 10, 2);
            n = Math.floor(n / 10);
        }
        // if sum exist, means it starts to cycle which does not include 1
        if (arr.indexOf(sum) !== -1) {
            return false;
        }
        // if sum is 1, return true
        if (sum === 1) {
            break;
        }
        arr.push(sum);
        n = sum;
        // need to reset sum everytime
        sum = 0;
    }
    return true;
};
